vuework04/src/store/index.js

```diff
@@ -2,41 +2,85 @@ import Vue from 'vue';
 import Vuex from 'vuex';
 import firebase from 'firebase/app';
 import 'firebase/auth';
+import 'firebase/firestore';
 
 Vue.use (Vuex);
 
 export default new Vuex.Store ({
   state: {
     users: [],
+    loginUser: {},
     error: null,
   },
+  getters: {
+    users: state => state.users,
+    loginUser: state => state.loginUser,
+    userName: state => (state.loginUser ? state.loginUser.displayName : ''),
+    uid: state => (state.loginUser ? state.loginUser.uid : null),
+  },
   mutations: {
-    addUser (state, {name, email, password}) {
-      const user = {
-        name,
-        email,
-        password,
-      };
+    addUser (state, user) {
       state.users.push (user);
+      console.log (state.users);
     },
-  },
-  getters: {
-    users (state) {
-      return state.users;
+    logIn({email, password}) {
+      email;
+      password;
+    },
+    setLoginUser (state, user) {
+      state.loginUser = user;
     },
   },
   actions: {
     addUser ({commit}, {name, email, password}) {
       firebase
         .auth ()
         .createUserWithEmailAndPassword (email, password)
-        .then (users => {
-          console.log (users);
-          alert ('Created user!!'), commit ('addUser', {
-            name,
-            email,
-            password,
-          });
+        .then (() => {
+          firebase
+            .auth ()
+            .currentUser.updateProfile ({
+              displayName: name,
+            })
+            .then (() => {
+              firebase
+                .firestore ()
+                .collection ('users')
+                .doc (email)
+                .set ({
+                  name: name,
+                  email: email,
+                  wallet: 500,
```

Comment:
キーとプロパティが一致しているときは次のように記述できますので修正しましょう。
```suggestion
                  name,
                  email,
                  wallet: 500,
```